File: C/mario-more/mario.c
// Make pyramids using hashes
// The user decides how tall the pyramids should be (between 1 and 8 inclusive)
// Problem link: https://cs50.harvard.edu/x/2022/psets/1/mario/more/

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8); //Check if input is between 1 and 8 inclusive

    // Print pyramid
    for (int i = 0; i < height; i++) // rows
    {
        // Space before pyramid
        for (int k = height - 1; k > i; k--)
        {
            printf(" ");
        }

        // First pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // Space between pyramids
        printf("  ");

        // Second pyramid
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: python/sentimental-mario-more/mario.py
# Problem link: https://cs50.harvard.edu/x/2022/psets/6/mario/more/

from cs50 import get_int

# Prompt the user for the pyramids’ height
height = get_int("Height: ")
# Check if it's a positive integer between 1 and 8, inclusive
# If the user fails to provide a positive integer between 1 and 8, re-prompt for the same again
while height < 1 or height > 8:
    height = get_int("Height: ")


# Generate pyramids

white_spaces = height - 1
hashes = 1

# Pyramids height
for i in range(height):
    # White spaces before pyramid
    for j in range(white_spaces):
        print(" ", end="")
    # Hashes for first pyramid
    for k in range(hashes):
        print("#", end="")
    # Two spaces between the two pyramids
    print("  ", end="")
    # Hashes for second pyramid
    for k in range(hashes):
        print("#", end="")
    white_spaces -= 1
    hashes += 1
    # Go to new line
    print("")